### ISSUE TITLE
Unsupported AllowedValues in parameter Cidr linter  

### ISSUE THREAD

**Issue by fatbasstard** [awslabs/cfn-python-lint#93]
The Paramter Cidr linter checks for an AllowedPattern, but it's also possible that a Cidr parameter is controlled with `AllowedValues`. This makes the check on the `AllowedPattern` obsolete:

https://github.com/awslabs/cfn-python-lint/blob/master/src/cfnlint/rules/parameters/Cidr.py#L65

Besides that the allowed values itself are not checked.

Example:

```
  EgressOracle:
    Type: "String"
    AllowedValues:
      - "XXX.XXX.XXX.XXX/28" # Testing
      - "YYY.YYY.YYY.YYY/28" # Production
    Description: "Egress (Outbound) CIDR for Oracle connectivity (port 1521)"
```

This raises the 
```
AllowedPattern for Parameter should be specified at Parameters/EgressOracle. Example "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$"
```

While there are AllowedValues that specify it. But those aren't checked, so this would be "OK" according to the current code:

```
  EgressOracle:
    Type: "String"
    AllowedValues:
      - "XXX.XXX.XXX.XXX" # Testing
      - "YYY.YYY.YYY.YYY" # Production
    Description: "Egress (Outbound) CIDR for Oracle connectivity (port 1521)"
```

There should be support for `AllowedValues`:
* Don't raise the AllowedPattern Warning if there are AllowedValues
* If there are AllowedValues, check those values for valid Cidr's

**Comment by fatbasstard** [comment]
@kddejong / @cmmeyer Can I be assigned to this issue? We're working on this, so PR will be coming soon.

(Just making issues of all remaining things that we encounter in our stacks to get a clear overview)

**Comment by kddejong** [comment]
makes sense to me.  thanks @fatbasstard
